src/Engine/ProtoCore/Lang/FunctionEndPoint.cs

```diff
@@ -309,15 +307,21 @@ public override string ToString()
                 sb.Append(FormalParams[FormalParams.Length - 1]);
             }
             sb.Append(")");
-            sb.Append("-> ");
-            sb.Append(returnType);
 
+            if (includeReturnType)
+            {
+                string returnType = procedureNode.ReturnType.ToString();
+
+                sb.Append("-> ");
+                sb.Append(returnType);
+            }
             return sb.ToString();
         }
-    
-    }
-
-
 
+        public sealed override string ToString()
```

Comment:
why seal this? Also, I am curious how this interacts with the unsealed base class ToString() - but mostly why? 😉

Reply:
Keeping in mind that it is a public method (which may already have 3d party derived classes....but hopefully not), I wanted to protect against Fragile base class (problem https://en.wikipedia.org/wiki/Fragile_base_class)
Using sealed in combination will not cause anything weird (I think). It just tells the compiler that no other class that derives from this specific one will be able to override the ToString() method.

Will remove this , does not make sense if I revert the changes here.